bug: Missing hypothesis package dependency causes test_properties.py to fail

## Description

The `hypothesis` Python package (property-based testing library) is missing from the project's dev dependencies, causing `test_properties.py` to fail with `ModuleNotFoundError: No module named 'hypothesis'` when running the test suite.

## Context

- **Project Area**: Testing infrastructure
- **Complexity Score**: 2/10 (simple dependency fix)
- **Discovered**: While running `pytest ergodic_insurance/tests/test_properties.py`

**Note**: There is an important distinction between:
- The `hypothesis` PyPI package - a property-based testing framework by David MacIver
- The project's `statistical_tests.py` - custom bootstrap-based statistical hypothesis testing

This issue concerns the missing `hypothesis` PyPI package.

## Technical Details

### Affected Files
| File | Issue |
|------|-------|
| `pyproject.toml` (root) | Missing `hypothesis` in `[project.optional-dependencies] dev` |
| `ergodic_insurance/pyproject.toml` | Missing `hypothesis` in `[project.optional-dependencies] dev` |
| `ergodic_insurance/tests/test_properties.py` | Imports `hypothesis` but package unavailable |

### Error Output
```
ImportError while importing test module 'ergodic_insurance\tests\test_properties.py'.
Traceback:
...
    from hypothesis import HealthCheck, assume, given, settings
E   ModuleNotFoundError: No module named 'hypothesis'
```

### What test_properties.py Tests
The file uses hypothesis for property-based testing of mathematical invariants:
- `TestErgodicProperties` - growth rate and ergodic coefficient properties
- `TestRiskMetricsProperties` - VaR monotonicity and TVaR relationships
- `TestLossDistributionProperties` - loss generation consistency
- `TestConvergenceProperties` - R-hat and ESS properties
- `TestOptimizationProperties` - optimization bounds

## Requirements

1. Add `hypothesis>=6.100.0` to dev dependencies in `pyproject.toml` files
2. Optionally add `hypothesis[numpy]` for numpy array strategies (already used via `hypothesis.extra.numpy`)
3. Ensure tests pass after installation

## Recommended Solution: add hypothesis as dev dependency

Add to both `pyproject.toml` files:
```toml
[project.optional-dependencies]
dev = [
    # existing dependencies...
    "hypothesis>=6.100.0",
]
```

## Implementation Steps

- [ ] Add `hypothesis>=6.100.0` to `pyproject.toml` (root) dev dependencies
- [ ] Add `hypothesis>=6.100.0` to `ergodic_insurance/pyproject.toml` dev dependencies
- [ ] Run `uv sync` or `pip install -e ".[dev]"` to install
- [ ] Verify tests pass with `pytest ergodic_insurance/tests/test_properties.py -v`
- [ ] Verify full test suite still passes

## Acceptance Criteria

- [ ] `pytest ergodic_insurance/tests/test_properties.py` runs without import errors
- [ ] All property-based tests pass
- [ ] Full test suite (`pytest`) passes
- [ ] Documentation updated if needed

--- a/ergodic_insurance/pyproject.toml
+++ b/ergodic_insurance/pyproject.toml
@@ -38,6 +38,7 @@ dev = [
     "mypy>=1.17.1",
     "isort>=6.0.1",
     "types-PyYAML>=6.0.0",
+    "hypothesis>=6.100.0",
 ]
 notebooks = [
     "jupyter>=1.1.1",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -42,6 +42,7 @@ dev = [
     "isort>=6.0.1",
     "types-PyYAML>=6.0.0",
     "pre-commit>=3.6.0",
+    "hypothesis>=6.100.0",
 ]
 notebooks = [
     "jupyter>=1.1.1",

